src/App.tsx

```diff
@@ -1,110 +1,135 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const options = [3, 5, 10, 20];
+
+interface ItemsListProps {
+  itemL: string[];
+}
+
+interface PaginationProps {
+  pageCount: number;
+  currentPage: number;
+  onPageChange: (page: number) => void;
+}
+
+export const ItemList: React.FC<ItemsListProps> = ({ itemL }) => (
+  <ul>
+    {itemL.map(item => (
+      <li key={item} data-cy="item">
+        {item}
+      </li>
+    ))}
+  </ul>
+);
+
+export const Pagination: React.FC<PaginationProps> = ({
+  pageCount,
+  currentPage,
+  onPageChange,
+}) => {
+  const pages = getNumbers(1, pageCount).map(n => n);
+
+  return (
+    <ul className="pagination">
+      <li className={`page-item ${currentPage === 1 ? 'disabled' : ''}`}>
+        <a
+          data-cy="prevLink"
+          className="page-link"
+          href="#prev"
+          aria-disabled={currentPage === 1}
+          onClick={() => currentPage > 1 && onPageChange(currentPage - 1)}
+        >
+          «
+        </a>
+      </li>
+
+      {pages.map(page => (
+        <li
+          key={page}
+          className={`page-item ${currentPage === page ? 'active' : ''}`}
+          onClick={() => onPageChange(page)}
+        >
+          <a data-cy="pageLink" className="page-link" href={`#${page}`}>
+            {page}
+          </a>
+        </li>
+      ))}
+      <li
+        className={`page-item ${currentPage === pageCount ? 'disabled' : ''}`}
+      >
+        <a
+          data-cy="nextLink"
+          className="page-link"
+          href="#next"
+          aria-disabled={currentPage === pageCount}
+          onClick={() =>
+            currentPage !== pageCount && onPageChange(currentPage + 1)
+          }
+        >
+          »
+        </a>
+      </li>
+    </ul>
+  );
+};
 
 export const App: React.FC = () => {
+  const [itemsPerPage, setItemsPerPage] = useState<number>(5);
+  const [currentPage, setCurrentPage] = useState<number>(1);
+
+  const pageCount: number = Math.ceil(items.length / itemsPerPage);
+  const startIndex = (currentPage - 1) * itemsPerPage;
+  const visibleItems = items.slice(startIndex, startIndex + itemsPerPage);
+
+  const handleItemsPerPageChange = (
+    event: React.ChangeEvent<HTMLSelectElement>,
+  ) => {
+    setItemsPerPage(parseInt(event.target.value));
+    setCurrentPage(1);
+  };
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        Page {currentPage} (items {startIndex + 1} -{' '}
+        {Math.min(startIndex + itemsPerPage, items.length)} of {items.length})
```

Comment:
The pagination info should be formatted as `Page {currentPage} (items {startIndex + 1} - {Math.min(startIndex + itemsPerPage, total)} of {total})`. Ensure the total number of items is used instead of `items.length`.